client/app/tasks/hooks/useEditTask.ts

```diff
@@ -0,0 +1,63 @@
+import { useRouter } from 'next/navigation'
+import { useQueryClient } from '@tanstack/react-query'
+import { useForm } from 'react-hook-form'
+import { useToast } from '@/_hooks'
+import { TaskType } from '@/_types'
+import { isErrorWithMessage } from '@/_utils'
+import { useEditTaskMutation } from './api'
+import { taskResolver, taskSchema } from '../schema'
+
+export const useEditTask = (defaultData: TaskType) => {
+  const queryClient = useQueryClient()
+  const { toast } = useToast()
+  const router = useRouter()
+
+  const { mutate, isPending } = useEditTaskMutation()
+
+  const form = useForm<taskSchema>({
+    resolver: taskResolver,
+    defaultValues: {
+      title: defaultData.title,
+      date: new Date(defaultData.date),
+      assigneeId: defaultData.assigneeId,
+      note: defaultData.note,
+    },
+  })
+
+  const handleSuccess = () => {
+    toast({ variant: 'success', title: 'Successfully edited a task!' })
+    queryClient.invalidateQueries({
+      // TODO: Add the query key "tasks" when you need
+      queryKey: ['task', defaultData.id],
+    })
```

Comment:
I found out that you can just same key for tasks and task.id. 
```.js
// useGetTaskQuery
    queryKey: ['tasks', { id: taskId }],
```

```.js
    queryClient.invalidateQueries({
      // TODO: Add the query key "tasks" when you need
      queryKey: ['tasks', { id: defaultData.id }],
    })
```

https://tanstack.com/query/v4/docs/framework/react/guides/query-invalidation

Reply:
Thanks! Fixed it!
https://github.com/HAK2024/fairy-share/pull/84/commits/d2e1b7191e4529b684796b264c74edc06e4476d9